osimpipeline/vital_tasks.py

```diff
@@ -482,39 +499,79 @@ def fill_setup_template(self, file_dep, target,
             content = content.replace('@MARKER_FILE@',
-                self.trial.marker_trajectories_fpath)
+                os.path.relpath(self.trial.marker_trajectories_fpath, 
+                    self.path))
             content = content.replace('@INIT_TIME@', '%.4f' % init_time)
             content = content.replace('@FINAL_TIME@', '%.4f' % final_time)
         
         with open(target[0], 'w') as f:
             f.write(content)
 
+
 class TaskIK(task.ToolTask):
     REGISTRY = []
     def __init__(self, trial, ik_setup_task, cycle=None):
-        super(TaskIK, self).__init__('ik', trial, cycle=cycle)
+        super(TaskIK, self).__init__(ik_setup_task, trial, cycle=cycle)
         self.doc = "Run OpenSim's Inverse Kinematics tool."
+        
         self.file_dep += [
                 self.subject.scaled_model_fpath,
                 ik_setup_task.results_tasks_fpath,
+                ik_setup_task.results_setup_fpath
                 ]
         self.targets += [
-                os.path.join(self.path, '%s_%s_ik_solution.mot' % (
-                    self.study.name, trial.id)),
-                os.path.join(self.path, 'ik_model_marker_locations.sto'),
+                ik_setup_task.solution_fpath,
+                ik_setup_task.model_markers_fpath
                 ]
 
+
+class TaskIKPost(task.PostTask):
+    REGISTRY=[]
+    def __init__(self, trial, ik_setup_task, cycle=None, error_markers=None):
+        super(TaskIKPost, self).__init__(ik_setup_task, trial, cycle=cycle)
+        self.doc = 'Create plots from the results of Inverse Kinematics.'
+        self.joint_angles_plotpath = '%s/joint_angles.pdf' % self.path
+        self.marker_errors_plotpath = '%s/marker_error.pdf' % self.path
+        self.error_markers = error_markers
+
+        self.add_action([ik_setup_task.solution_fpath],
+                        [self.joint_angles_plotpath],
+                        self.joint_angle_plots)
+
+        if self.error_markers:
+            self.add_action([self.subject.scaled_model_fpath, 
+                            ik_setup_task.model_markers_fpath,
+                            self.trial.marker_trajectories_fpath],
+                            [self.marker_errors_plotpath],
+                            self.marker_error_plots)
+
+    def joint_angle_plots(self, file_dep, target):
+        # if os.path.exists(self.fig_fpath):
+        #     os.rename(self.fig_fpath,
+        #             self.fig_fpath.replace('.pdf', '_backup.pdf'))
+        fig = plot_lower_limb_kinematics(file_dep[0], self.gl)
+        fig.savefig(target[0])
+        pl.close(fig)
+
+    def marker_error_plots(self, file_dep, target):
+        # if os.path.exists(self.errorplot_fpath):
+        #     os.rename(self.errorplot_fpath,
+        #             self.errorplot_fpath.replace('.pdf', '_backup.pdf'))
+        plot_marker_error(target[0], self.error_markers, 
+            10, self.gl, file_dep[0], file_dep[1], file_dep[2])
```

Comment:
I like that you're keeping a cleaner separation of tasks and plotting functions.